src/olympia/addons/forms.py

```diff
@@ -208,21 +208,19 @@ class CategoryForm(forms.Form):
         queryset=Category.objects.all(), widget=CategoriesSelectMultiple)
 
     def save(self, addon):
-        application = self.cleaned_data.get('application')
-        categories_new = self.cleaned_data['categories']
-        categories_old = [cats for app, cats in addon.app_categories if
-                          (app and application and app.id == application) or
-                          (not app and not application)]
-        if categories_old:
-            categories_old = categories_old[0]
+        application = self.cleaned_data.get('application', amo.FIREFOX.id)
```

Comment:
Hopefully nowhere, but it's possible since it's not `required`. The old fallback was `None`, which is not great. Re-reading this though, I should replace `addon.app_categories[]` below by add a `addon.app_categories.get()` instead - the goal here is simply to avoid failing catastrophically if `application` is absent.